src/math_functions.cpp

```diff
@@ -674,6 +674,35 @@ double maxwell_spectrum(double T) {
 }
 
 
+double gaussian_spectrum(double mean, double standard_deviation) {
+  double r2 = 1.01;
+  double v1 = 0.;
+  double v2 = 0.;
+  
+  // perhaps there should be a limit to the number of resamples
+  while ( r2 > 1 ) {
```

Comment:
Since the `return` statement is inside, there's no need to have a condition here. Just make this `while (true) {`